Terminal.Gui/Drivers/V2/UnixKeyConverter.cs

```diff
@@ -0,0 +1,20 @@
+﻿#nullable enable
+
+namespace Terminal.Gui.Drivers;
+
+/// <summary>
+///     <see cref="IKeyConverter{T}"/> capable of converting the
+///     unix native <see cref="char"/> class
+///     into Terminal.Gui shared <see cref="Key"/> representation
+///     (used by <see cref="View"/> etc).
+/// </summary>
+internal class UnixKeyConverter : IKeyConverter<char>
+{
+    /// <inheritdoc />
+    public Key ToKey (char value)
+    {
+        ConsoleKeyInfo adjustedInput = EscSeqUtils.MapChar (value);
+
+        return EscSeqUtils.MapKey (adjustedInput);
```

Comment:
Does operating purely on char cause issues later on for surrogate pairs? Or is ok?

Reply:
Absolutely works as expected because like all the drivers surrogate pairs inputs are always considered the join of two chars (high and low surrogates). You can copy and paste by keyboard or context menu a surrogate pair in a TextField or TextView without any issue.